### PR TITLE
Hide masterbar in webviews

### LINKED ISSUE #6504: Hide the masterbar in web views.
When clicking "View Site", "View Post", or any other link that opens my wpcom site in a web view, Calypso's masterbar is visible. Because the icons are the same as the ones on our tab bar, the user could easily accidentally navigate to other parts of Calypso, causing significant confusion. For new users, the guided overlays that appear on the web could increase the chances that these accidental clicks happen.

We can append a query string to links that open webviews to hide the Calypso masterbar. It's as follows: `?preview=true&iframe=true` (thanks @folletto) ([Calypso ref](https://github.com/Automattic/wp-calypso/blob/47f1a198578f9827f73e42539baa4f112de1fac4/client/my-sites/themes/helpers.js#L20))

In my testing, I think adding this query string will both hide the masterbar, and prevent the guided overlay for new users from appearing, when in a web view. We'd want to test this with existing and new accounts to make sure it resolves the issue completely.

I haven't looked at what the situation is with self-hosted, non-Jetpack sites; we need to figure out if they see the core adminbar or not, and whether there's anything we can do about those cases.

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #6504 

This hides the WordPress.com master bar and Jetpack admin bar in `WebView`s. 

**Master bar:**
masterbar

**Admin bar:**
adminbar

As noted on #6504 , the admin bar will still show for self-hosted (non-JP) sites.

**To test:**
There are several places that display a `WebView`, but since most are handled by `WebViewControllerFactory` they will all behave the same way. So I'll just list a few:
- [x] WP site > View Site.
- [x] JP site > View Site.
- [x] WP site > Blog Posts > View on a post.
- [x] JP site > Blog Posts > View on a post.
- [x] Me > App Settings > Privacy Settings > all links
- [x] Also, please test on iOS 11 and iOS 10 as there are different controllers depending on iOS version.

### CODE REVIEW

File: WordPress/Classes/Extensions/URL+Helpers.swift [hunk 1]
```diff
@@ -115,6 +115,17 @@ extension URL {
             return pathExtension.lowercased() == "gif"
         }
     }
+
+    func appendHideMasterbarParameters() -> URL? {
```
Comment (frosty) [inline WordPress/Classes/Extensions/URL+Helpers.swift]: I'd personally rename this slightly, as the current name (to me) implies that it's going to modify the URL that you call it on, whereas it's actually returning a new URL. Perhaps `appendingHideMasterbarParameters` or `urlWithAppendedMasterbarParameters`?

File: WordPress/Classes/Extensions/URL+Helpers.swift [hunk 2]
```diff
@@ -128,4 +139,9 @@ extension NSURL {
         }
         return NSNumber(value: fileSize)
     }
+
+    @objc func appendHideMasterbarParameters() -> NSURL? {
```
Comment (frosty) [inline WordPress/Classes/Extensions/URL+Helpers.swift]: Same as the earlier comment about naming :)

File: WordPress/Classes/Utility/WebKitViewController.swift [hunk 3]
```diff
@@ -104,6 +107,13 @@ class WebKitViewController: UIViewController {
         if addsWPComReferrer {
             request.setValue("https://wordpress.com", forHTTPHeaderField: "Referer")
         }
+
+        if addsHideDotComMasterbarParameters,
+            let host = request.url?.host,
+            host.contains("wordpress.com") {
```
Comment (frosty) [inline WordPress/Classes/Utility/WebKitViewController.swift]: Should we perhaps extract a constant for `"wordpress.com"`?

File: WordPress/Classes/Utility/WebViewControllerConfiguration.swift [hunk 4]
```diff
@@ -3,8 +3,9 @@ import UIKit
 class WebViewControllerConfiguration: NSObject {
     @objc var url: URL
     @objc var optionsButton: UIBarButtonItem?
-    @objc var secureInteraction: Bool = false
-    @objc var addsWPComReferrer: Bool = false
+    @objc var secureInteraction = false
```
Comment (frosty) [inline WordPress/Classes/Utility/WebViewControllerConfiguration.swift]: Yay for type inference 👍

**Comment by ScoutHarris** [comment]
> I'm still seeing the master bar / admin bar when I tap on any of the links in the Privacy section.

Interesting. It seems to hide fine in iOS11, but on iOS10. Thanks for finding that!

> I guess it's still expected that the bottom bar is visible?

Yep, this is only for the masterbar.

**Comment by ScoutHarris** [comment]
OK @frosty could you give it another look please? Thanks!

**Comment by ScoutHarris** [comment]
Hey @frosty . Good point as always. I moved the URLs to `Constants` where other URLs live.

**Comment by alisterscott** [comment]
I tested this as part of the WPiOS 10.3.0.20180618 on iPhone 7 Plus iOS 11.4 and I can still see the masterbar for a WordPress.com site when viewing a post or a page (from the posts/pages list) but **not** when using view site.

Is this expected in this release to now show for post/page views?

view post page site - 1

view post page site - 2

**Comment by ScoutHarris** [comment]
Hey @alisterscott . Thank you for finding this. I'll look at it for the next release.

**Pull request merged**